src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+// Disable revive linter since any structs in this file will expose the same
+// funcs as the original structs in the msgraph-sdk-go package, which do not
+// follow some of the golint rules.
+//
+//nolint:revive
+package custom
+
+import (
+	"strings"
+	"time"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+// Replica of models.DriveItemable
+type LiteDriveItemable interface {
+	GetId() *string
+	GetName() *string
+	GetSize() *int64
+	GetFolder() interface{}
+	GetPackageEscaped() interface{}
+	GetShared() interface{}
+	GetMalware() malwareable
+	GetDeleted() interface{}
+	GetRoot() interface{}
+	GetFile() fileItemable
+	GetParentReference() itemReferenceable
+	SetParentReference(itemReferenceable)
+	GetCreatedBy() identitySetable
+	GetCreatedByUser() userable
+	GetLastModifiedByUser() userable
+	GetCreatedDateTime() *time.Time
+	GetLastModifiedDateTime() *time.Time
+	GetAdditionalData() map[string]interface{}
+}
+
+var _ LiteDriveItemable = &driveItem{}
+
+type driveItem struct {
+	id                   string
```

Comment:
Can these values be exported?  I know we're abiding the getter interface, but it at least seems like a shortcut to struct creation.

Reply:
Just to elaborate on this, if we want to set properties on this `driveItem` struct, we can first create a `models.NewDriveItem()` and then call `ToLiteDriveItemable` on it. We could have a shortcut like you said, but there is no need for it, so I think we can skip it for now.

Reply:
That seems like a lot of roundabout just to populate a struct.

Reply:
There is no current need to directly populate this struct anywhere. We can export the fields if a need arises in future.

Reply:
Okedoke!  Sounds good.